core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java

```diff
@@ -2139,6 +2141,7 @@ else if (sourceNode instanceof SampleNode sampleNode) {
                         OptionalInt.empty());
 
                 if (columns != null) {
+                    verify(columnTypes != null);
```

Comment:
this won't produce any useful message in case it's null
if you _don't_ verify explicitly here, the exception message will contain "columnTypes".